test/specs/templates.e2e.ts

```diff
@@ -0,0 +1,283 @@
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Push Source to Default Scratch Org', 5);
+
+    const successNotificationWasFound = await utilities.attemptToFindNotification(workbench, 'SFDX: Push Source to Default Scratch Org successfully ran', 10);
+    expect(successNotificationWasFound).toBe(true);
+
+    // Check the output.
+    const outputPanelText = await utilities.attemptToFindOutputPanelText('Salesforce CLI', '=== Pushed Source', 10);
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('Created  auraEvent1  AuraDefinitionBundle');
+  });
+
+  // Apex Class
+  step('Create an Apex Class', async () => {
+    // Using the Command palette, run SFDX: Create Apex Class.
+    const workbench = await browser.getWorkbench();
+    const inputBox = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Create Apex Class', 1);
+
+    // Set the name of the new component to ApexClass1.
+    await inputBox.setText('ApexClass1');
+    await inputBox.confirm();
+    await utilities.pause(1);
+
+    // Select the default directory (press Enter/Return).
+    await inputBox.confirm();
+
+    const successNotificationWasFound = await utilities.attemptToFindNotification(workbench, 'SFDX: Create Apex Class successfully ran', 10);
+    expect(successNotificationWasFound).toBe(true);
+
+    const outputPanelText = await utilities.attemptToFindOutputPanelText('Salesforce CLI', 'Finished SFDX: Create Apex Class', 10);
+    expect(outputPanelText).not.toBeUndefined();
+
+    // Check for expected items in the Explorer view.
+    const sidebar = workbench.getSideBar();
+    const treeViewSection = await sidebar.getContent().getSection(projectName);
+    await treeViewSection.expand();
+
+    // Get the matching (visible) items within the tree which contain "apexClass1".
+    const filteredTreeViewItems = await utilities.getFilteredVisibleTreeViewItemLabels(workbench, projectName, 'ApexClass1');
+
+    // It's a tree, but it's also a list.  Everything in the view is actually flat
+    // and returned from the call to visibleItems.reduce().
+    expect(filteredTreeViewItems.includes('ApexClass1.cls')).toBe(true);
+    expect(filteredTreeViewItems.includes('ApexClass1.cls-meta.xml')).toBe(true);
+  });
+
+  step('Verify the contents of the Apex Class', async () => {
+    const expectedText = [
+      'public with sharing class ApexClass1 {',
+      '    public ApexClass1() {',
+      '',
+      '    }',
+      '}'
+    ].join('\n');
+    const workbench = await browser.getWorkbench();
+    const editorView = await workbench.getEditorView();
+    const textEditor = await editorView.openEditor('ApexClass1.cls') as TextEditor;
+    const textGeneratedFromTemplate = (await textEditor.getText()).trimEnd();
+    expect(textGeneratedFromTemplate).toEqual(expectedText);
+  });
+
+  step('Push the Apex Class', async () => {
```

Comment:
Since we're testing push/pull separately, does this feel duplicative?

Reply:
Discussed this and since it's pushing a few different types, that's the value add here.

Reply:
I think it's OK to have some overlap.  The purpose of templates.e2e.ts is to test the various template features around the three data types (Apex, Aura, and LWC) and as part of the suite the code generated via a template is pushed.  So in this file, various template features are exercised, but push & pull only minimally.

Contrary to this, the purpose of pushAndPull.e2e.ts is to test functionality around push and pull, and so it go deeper within these commands. (and deeper, once conflict detection has been added)